requests.go

```diff
@@ -160,5 +160,9 @@ func (c *Client) put(path string, stream io.Reader) int {
 
 func (c *Client) createParentCollection(itemPath string) (err error) {
 	parentPath := path.Dir(itemPath)
+	if parentPath == "." {
```

Comment:
You should also check for root path ('/') here. Add it and your pull request will be accepted.